modules/Microsoft.Sql/servers/.test/secondary/deploy.test.bicep

```diff
@@ -0,0 +1,64 @@
+targetScope = 'subscription'
+
+// ========== //
+// Parameters //
+// ========== //
+
+@description('Optional. The name of the resource group to deploy for testing purposes.')
+@maxLength(90)
+param resourceGroupName string = 'ms.sql.servers-${serviceShort}-rg'
+
+@description('Optional. The location to deploy resources to.')
+param location string = deployment().location
+
+@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints.')
+param serviceShort string = 'sqlsec'
+
+@description('Optional. The password to leverage for the login.')
+@secure()
+param password string = newGuid()
+
+
+// ============ //
+// Dependencies //
+// ============ //
+
+// General resources
+// =================
+resource resourceGroup 'Microsoft.Resources/resourceGroups@2021-04-01' = {
+  name: resourceGroupName
+  location: location
+}
+
+module nestedDependencies 'dependencies.bicep' = {
+  scope: resourceGroup
+  name: '${uniqueString(deployment().name, location)}-nestedDependencies'
+  params: {
+    serverName: '<<namePrefix>>-${serviceShort}-pri'
+    databaseName: 'db1'
```

Comment:
Can be moved as hardcoded into the `dependencies.bicep`.

If the name can be anything (i.e. does not conflict with anything else) we don't need to apply the `servicesShort` and hence can directly use it in the file.

We do something similar for, for example, subnet names. The virtual network name is passed in like the `serverName` above, but the subnet is hardcoded as `name: 'defaultSubnet'` directly in the `dependencies.bicep` file.

Reply:
The name of the server needs to be globally unique, but the database name has no such restrictions. I can hardcore the database name into the dependencies.

Reply:
Note you may need to introduce a variable though, as you use the name in 2 locations. I'll add a suggestion too, to use a native bicep resource (in the comment that queries @eriqua)